pkg/geo/parse.go

```diff
@@ -25,6 +26,122 @@ import (
 	"github.com/twpayne/go-geom/encoding/wkb"
 )
 
+// Threshold to assume that both float64 values are
+// equivilent
+const float64EquivalentThreshold = 1e-9
+
+// longitudeNormalize will normalize angles to
+// spherical polar angles (-180 to 180)
+func longitudeNormalize(log float64) float64 {
+	if log > 360 {
+		log = math.Remainder(log, 360)
+	}
+	if log < -360 {
+		log = math.Remainder(log, -360)
+	}
+	if log > 180 {
+		log += -360
+	}
+	if log < -180 {
+		log += 360
+	}
+	if log == -180 {
+		return 180
+	}
+	if log == -360 {
+		return 0
+	}
+	return log
+}
+
+// latitudeNormalize will normalize angles to
+// spherical azimuth angle (-90 to 90)
+func latitudeNormalize(lat float64) float64 {
+	if lat > 360 {
+		lat = math.Remainder(lat, 360)
+	}
+	if lat < -360 {
+		lat = math.Remainder(lat, -360)
+	}
+	if lat > 180 {
+		lat = 180 - lat
+	}
+	if lat < 180 {
+		lat = -180 - lat
+	}
+	if lat > 90 {
+		lat = 180 - lat
+	}
+	if lat < -90 {
+		lat = -180 - lat
+	}
+
+	return lat
+}
+
+func makeValidPoint(lat float64, log float64) (float64, float64) {
+	if math.IsNaN(lat) || math.IsNaN(log) {
+		return lat, log
+	}
+	_lat := latitudeNormalize(lat)
+	_log := longitudeNormalize(log)
```

Comment:
we usually do not use `_` in Golang -- this should probably be a different name.